docs/documentation/head/server-prepare.md

```diff
@@ -9,37 +9,143 @@ nexttitle: Physical and Logical replication API
 next: replication.html
 ---
 
+### Motivation
+
 The PostgreSQL™ server allows clients to compile sql statements that are expected
 to be reused to avoid the overhead of parsing and planning the statement for every
 execution. This functionality is available at the SQL level via PREPARE and EXECUTE
 beginning with server version 7.3, and at the protocol level beginning with server
 version 7.4, but as Java developers we really just want to use the standard
 `PreparedStatement` interface.
 
-### Note
+> Server side prepared statements are planned only once by the server. This avoids
+the cost of replanning the query every time, but also means that the planner
+cannot take advantage of the particular parameter values used in a particular
+execution of the query. You should be cautious about enabling the use of server
+side prepared statements globally.
+
+Server side prepared statements can improve execution speed as
+1. It sends just statement handle (e.g. `S_1`) instead of full SQL text
+1. It enables use of binary transfer (e.g. binary int4, binary timestamps, etc), and it is much faster to parse
```

Comment:
I would say replace ", and it..." with "; the parameters and results are much faster to parse"